File: src/main/java/io/openfuture/openmessanger/web/controller/ChatController.java
package io.openfuture.openmessanger.web.controller;

import org.springframework.messaging.handler.annotation.MessageMapping;
import org.springframework.messaging.handler.annotation.Payload;
import org.springframework.stereotype.Controller;

import io.openfuture.openmessanger.service.MessageService;
import io.openfuture.openmessanger.web.request.MessageRequest;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;

@Slf4j
@Controller
@RequiredArgsConstructor
public class ChatController {

    private final MessageService messageService;

    @MessageMapping("/direct-message")
    public void sendMessage(@Payload MessageRequest request) {
        log.info("REQUEST {}", request);
        messageService.sendMessage(request);
    }

}

File: src/main/kotlin/io/openfuture/openmessanger/web/controller/ChatController.kt
package io.openfuture.openmessanger.web.controller

import io.openfuture.openmessanger.service.MessageService
import io.openfuture.openmessanger.web.request.MessageRequest
import org.springframework.messaging.handler.annotation.MessageMapping
import org.springframework.messaging.handler.annotation.Payload
import org.springframework.stereotype.Controller

@Controller
class ChatController(
    val messageService: MessageService
) {

    @MessageMapping("/direct-message")
    fun sendMessage(@Payload request: MessageRequest) {
        messageService.sendMessage(request)
    }

}